drivers/ninaw10/machine_pin_nina.c

```diff
@@ -42,8 +42,10 @@
 
 #define NINA_GPIO_MODE          (0x50)
 #define NINA_GPIO_READ          (0x53)
+#define NINA_GPIO_READ_ANALOG   (0x54)
 #define NINA_GPIO_WRITE         (0x51)
-#define NINA_GPIO_IS_INPUT_ONLY(p) ((p >= 34 && p <= 36) || (p == 39))
+#define NINA_GPIO_IS_INPUT_ONLY(p)  ((p >= 34 && p <= 36) || (p == 39))
+#define NINA_GPIO_IS_ADC_CHANNEL(c) ((c >= 3 && c <= 6))
```

Comment:
The `NINA_GPIO_IS_INPUT_ONLY` and `NINA_GPIO_IS_ADC_CHANNEL` macros currently take different argument types, yet they are named as though they take the same type.

As far as I can tell there are 3 distinct numbering schemes used in this file:
1. The id of a pin as seen by Python, with range 0 to 6 inclusive, that's what `pin->id` is.  `NINA_GPIO_IS_ADC_CHANNEL` works with these numbers.
2. The gpio number as known by the Nina, range 25 through 39.  `NINA_GPIO_IS_INPUT_ONLY` takes in this number and tells if that gpio is input only.
3. The ADC channel known by the Nina.

`pin_map` maps 1 to 2.

`adc_map` maps 1 to 3.

I think `NINA_GPIO_IS_ADC_CHANNEL` should be renamed to `NINA_GPIO_HAS_ADC_CAPABILITIES` and work on gpio numbers of type 2, ie it would validate the argument is between 34 and 39.

Reply:
> I think NINA_GPIO_IS_ADC_CHANNEL should be renamed to NINA_GPIO_HAS_ADC_CAPABILITIES and work on gpio numbers of type 2, ie it would validate the argument is between 34 and 39.

That would require mapping the logical `self->id` to physical pin id first, I think it would be easier to just use the logical pin id everywhere. That would make the two macros the same, however I still would like to keep 2 separate macros, it's more clear this way.

Reply:
> however I still would like to keep 2 separate macros, it's more clear this way

Then maybe call this new macro something like `NINA_LOGICAL_PIN_HAS_ADC`?

Reply:
Both macros now accept logical pin ID (as seen by the outside code), I don't think there's a point in differentiating this macro from the other.
```
#define NINA_GPIO_IS_INPUT_ONLY(p)  ((p >= 3 && p <= 6))
#define NINA_GPIO_IS_ADC_CHANNEL(p) ((p >= 3 && p <= 6))
```

Reply:
Yeah, this is much better now.